pageserver/src/layered_repository/delta_layer.rs

```diff
@@ -181,12 +181,20 @@ impl Layer for DeltaLayer {
         &self,
         blknum: u32,
         lsn: Lsn,
+        cached_img_lsn: Option<Lsn>,
         reconstruct_data: &mut PageReconstructData,
     ) -> Result<PageReconstructResult> {
         let mut need_image = true;
 
         assert!(self.seg.blknum_in_seg(blknum));
 
+        match &cached_img_lsn {
+            Some(cached_lsn) if &self.end_lsn <= cached_lsn => {
```

Comment:
Is this right?

Reply:
Yeah, sounds right. I wonder why the caller would call this function in the first place, though, if it has a cached image at or after this layer's end LSN?